web/modules/custom/dpl_consumers/src/Commands/DplConsumersCommands.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\dpl_consumers\Commands;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\dpl_consumers\DplGraphqlConsumersConstants;
+use Drush\Commands\DrushCommands;
+
+/**
+ * Drush commands for DPL consumers.
+ */
+final class DplConsumersCommands extends DrushCommands {
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entityTypeManager,
+  ) {
+    parent::__construct();
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * Function is used for setting the consumer secret.
+   *
+   * @command dpl_consumers:set-consumer-secret
+   */
+  public function setConsumerSecret(): string {
+    try {
+      /** @var \Drupal\Core\Entity\EntityStorageInterface $consumer */
+      $consumer = $this->entityTypeManager
+        ->getStorage('consumer')
+        ->loadByProperties(['client_id' => DplGraphqlConsumersConstants::GRAPHQL_CONSUMER_CLIENT_ID]);
+
+      $consumer = reset($consumer);
+
+      $secret = getenv('GRAPHQL_CONSUMER_SECRET');
+      if (!empty($secret)) {
+        $consumer->secret = $secret;
+        $consumer->save();
+
+        return 'Consumer secret set successfully.';
+      }
+
+      return 'Consumer secret was not found.';
+    }
+    catch (\Exception $e) {
+      return 'Error fetching consumer.';
+    }
+  }
```

Comment:
We should not mix errors and result.
If errors occur we should throw an exception

Reply:
Please take another look, I have exchanged the "faulty" return strings for exceptions instead.